### ISSUE TITLE
[P0][Contracts] スコア型の二重定義を統合し、未校正の確率受理を防ぐ

### ISSUE THREAD

**Issue by dj-thank** [dj-thank/semantic-asr#25]
親: #23。依存: なし（他の型移行より先行）。監査基準 `c051e397`。

## 確認事実
`score_semantics.py::EvidenceScore` と `score_types.py::EvidenceScore` は同名だが別型・別validation。前者で `EvidenceScore(0.9, ScoreKind.PROBABILITY, 'fixture', calibrated=True).require_probability()` がcalibration_digestなしで0.9を返すことを再現。後者は平均／累積尤度を区別するが校正来歴のチェックを合わせる必要がある。名前の一致だけで置換しない。
読む: 上記2ファイル、`calibration.py`, `ranker_calibration.py`, `deliberation_evidence.py`, `sequence_scorers.py`, `fusion.py`。

## 実装手順
1. 全producer/consumerを列挙し、意味・単位・正規化・score domain・欠損・校正資料を対応表にする。
2. canonical score契約を1つに決めるADRを追加。平均と累積の尤度、logit、preference、bounded utility、correctness probabilityを区別。frame/token正規化の違いを失わない。
3. 旧importからの互換adapterを設け、変換が情報を失う場合は明示エラー。バージョン付きserialization fixtureを用意してから呼出箇所を一つずつ移行。
4. 「校正済み」として利用するには、型だけでなく、存在する凍結校正profileと一致するdigest・score source・入力条件・データsplitを検査。64文字の自己申告だけでは校正を証明しない。
5. 数値validationとcanonical hashを共有化。ただしhash payloadを変える場合はschema migrationを付ける。

## 受入条件
- [ ] digestなし／未知profile／別model／別score kind／別正規化の校正主張は拒否。
- [ ] bool、NaN、Inf、不正enum、mutable metadata変更を否定テスト。
- [ ] 同一domain内のみ集約、異なるモデル・区間・prompt・温度の分布を無断で足さない。
- [ ] legacy serialization round-tripと新schema移行goldenが通る。既存8件の公開判定replayも維持。
- [ ] `tests/test_score_contract_migration.py`（新規）で上記を実行し、`python scripts/replay_phonetic_decisions.py` と全suite成功。

## 禁止・完了証拠
rawなLLM回答をsigmoidで0–1にするだけで確率にしない。全てをfloatに戻す統合や、複雑な汎用framework化は不可。ADR・呼出一覧・RED/GREEN・移行表・serialized fixtureをPRへ。正しさの修正で判定が変わる場合は旧結果との差分を説明し、精度向上と混同しない。

**Comment by dj-thank** [comment]
着手記録です。

Base: `5eba248ca85c7c87923b2016cd0d2a764f4a28f4`
Tree: `4837f1ffac0c0e1aad4a296f0d731dc9e348c5dd`
Branch: `codex/canonical-score-contract-25`

編集予定:
- `src/semantic_asr/score_contract.py`（canonical型・profile registry・serialization/migration）
- `src/semantic_asr/score_types.py`（canonical型の再exportとcalibrator接続）
- `src/semantic_asr/score_semantics.py`（旧import互換shim）
- `src/semantic_asr/phonetic_evidence.py`
- `src/semantic_asr/deliberation_evidence.py`
- `src/semantic_asr/calibration.py` / `fusion.py`（暗黙の[0,1]確率受理を廃止）
- `src/semantic_asr/ranker_calibration.py`
- `src/semantic_asr/__init__.py`
- `tests/test_score_contract_migration.py` と既存関連tests
- ADR、producer/consumer inventory、versioned golden fixtures

先に現在の未校正確率受理・未知profile・別model/kind/normalization・mutable metadataをREDで固定し、旧serialization replayを残したまま段階移行します。認識精度向上とは分離した契約修正です。

**Comment by dj-thank** [comment]
## Completion evidence

Issue #25 is implemented on `main` by merge commit `3d998afdfff460c3abcebedb9a4aec44a677147a`.

Exact main tree:

```text
5bfef190a98b043614a6a4a9f1df6e651833650f
```

The merged tree is byte-identical to the synthetic merge tree independently validated before merge.

Post-merge push workflows are all green on the main commit:

- Semantic ASR CI #294
- Frontier contracts #228
- PCM/runtime integrity #39
- Reproducible validation evidence #42

This includes Ubuntu Python 3.11/3.12, Windows Python 3.12, Ruff format/lint, compileall, clean wheel installation, model-free tests/demos, optional CPU PyTorch training/acoustic tests, Ubuntu/Windows PCM and orchestration integrity, research registry, workflow parsing, and tracked-source immutability.

The exact pre-merge merge-source artifact was SHA-256 verified and independently executed under Python 3.13.5: `639 passed`; all 8 stored phonetic decisions replayed; all public `EvidenceScore` imports resolve to the single implementation.

Completion scope: canonical score semantics, normalization, provenance, score-domain aggregation, immutable metadata, versioned migration, and registry-authorized correctness probabilities. No recognition-quality or default-profile claim is attached to this completion.
